sys/include/net/nanocoap.h

```diff
@@ -2219,10 +2369,21 @@ static inline coap_method_flags_t coap_method2flag(unsigned code)
  * @param[in]   buf     pointer to raw packet data
  * @param[in]   len     length of packet at @p buf
  *
- * @returns     0 on success
- * @returns     <0 on error
+ * @return      Number of bytes parsed (may not match @p len on stream
+ *              transports)
```

Comment:
This API change is actually not so benign.
Previously 0 would mean success, so code would check `if (coap_parse(…)) {}` as the *error* case.

Now we need to check `if (coap_parse(…) <= 0) {}`

This should at least be featured more prominently if no API compatibility is possible.

Reply:
This is indeed the main breaking change and highlighted as such in the PR description.

I think the fallout is pretty limited, though: User code will not really parse CoAP packets by hand, but rather use the CoAP server implementations and provide the resource handlers instead, which already will be called on fully parsed packets